### ISSUE TITLE
Memory Leak in decode if claim validation fails - claims with length zero is not freed.

### ISSUE THREAD

**Issue by gajanak** [GlitchedPolygons/l8w8jwt#46]
Hello,

At first, thank for this very usefull library.

My Finding:

On decoding(in l8w8jwt_decode)  chillbuff_init of 'claim' is called - allocating memory
Later ... 'claim'  is assigned/transfered to out_claims pointer.
later ... if out_claims is not null - claims is not freed.
problem is that claimlength is zero.
so a to l8w8jwt_free_claims(claims,claimslength) if claims is pointer and length is zero does nothing -->
Leak ;)

Simple Workaround: Change Line 841 - out_claims will only be set, if there are any claims in chilbuf.
so empty chilbuf is freed in decode.

https://github.com/GlitchedPolygons/l8w8jwt/blob/a7afbfcf2ea5b5d52ffc613582d271b881e7df8f/src/decode.c#L841

```patch
- if (out_claims != NULL && out_claims_length != NULL)
+ if (out_claims != NULL && out_claims_length != NULL && claims.length>0)
```
 
 Or changing l8w8jwt_free_claims to free claim if claims_count==0 ....
 
 PS: r was  L8W8JWT_SUCCESS - because only exp validation fails.

**Comment by GlitchedPolygons** [comment]
@gajanak oh, thank you so much for this. Good catch!
I'll look into it as soon as I can. Thank you for making l8w8jwt more stable 🙏🏻

**Comment by GlitchedPolygons** [comment]
@gajanak Fixed it via bb1b07c - is that what you had in mind? :)

Will package+sign a release as soon as I can 👍🏻
